src/client/Microsoft.Identity.Client/MsalServiceException.cs

```diff
@@ -147,10 +151,15 @@ public MsalServiceException(string errorCode, string errorMessage, int statusCod
         /// http://msdn.microsoft.com/en-us/library/bb268233(v=vs.85).aspx).
         /// You can use this code for purposes such as implementing retry logic or error investigation.
         /// </summary>
-        public int StatusCode { get; internal set; } = 0;
-
-#if !DESKTOP
-#endif
+        public int StatusCode
```

Comment:
This is such a nasty bug I found here. `StatusCode` can be updated, and in fact the exception factory does not use the ctor to create this exception. But when `StatusCode` is updated (e.g. from 0 to 502), the `IsRetryable` property is not updated. 

Note to self: never trust mutable objects.